### DISCUSSION TITLE
Including Libraries in Elixir Kata

### DISCUSSION THREAD

**Discussion by bellmyer** [codewars/codewars.com#3003] [General]
JSON manipulation is a big part of backend/API coding, and I'm not seeing any Elixir kata that deal with this. Even trying to create some, I can't find a way to include the JSON module in the code. Does CodeWars have a way to include libraries that would normally be included via the `mix` utility?

I don't understand the underlying mechanics of how elixir is safely compiled/run within CodeWars. I understand this may create security concerns, but it seems like there are several libraries (like JSON) which could be preloaded and available for import. 

Perhaps this is already the case, but I don't have the syntax right. If so, I'd appreciate some guidance there, too. I did my best to google and search through other kata to find my answer before asking here.

**Comment by Molksy** [comment]
CodeWars does not currently support including external libraries or dependencies in Elixir kata. This is due to security concerns, as allowing arbitrary code execution can open up potential vulnerabilities.

However, there are ways to work with JSON in Elixir without the need for external libraries. Elixir provides a built-in module called Jason for working with JSON. This module is part of Elixir's standard library and does not require any additional setup or installation.

To use the Jason module, you can simply include it in your Elixir code by adding the following line at the top of your file:

"require Jason"

You can then use the functions provided by the Jason module to parse and generate JSON data.

Here's an example of how you can use the Jason module to parse a JSON string:

"json_string = "{\"name\": \"John\", \"age\": 30}"
json_data = Jason.decode!(json_string)"

In this example, we're using the decode!/1 function provided by the Jason module to parse the JSON string into an Elixir map. The ! at the end of the function name indicates that it will raise an error if the JSON string is not valid.

**Reply by bellmyer** [reply]
Thank you for your response. However, it doesn't look like Jason (or JSON, or Json) are require-able within katas. The Jason module itself doesn't appear to be part of the Elixir core.

I appreciate the security issues involved with a system like this. But I think there's definitely a place for making some pre-approved modules available, such as JSON. Otherwise, languages with thin cores are really limited on this platform. And as a result, the higher level kata are merely more and more complicated math/string puzzles, not really building usable skills in the language itself.

FWIW, the supported version of elixir (1.11) is reaching full end-of-life (no more security support) in a couple weeks. If elixir isn't worth supporting, that's codewars' call. I just wish we were able to do more with the language on this amazing platform.

**Reply by Blind4Basics** [reply]
@bellmyer considering the [other answer](https://github.com/codewars/codewars.com/discussions/3001#discussioncomment-5904447) of this user, I'd suggest to take anything they say as doubtful until confirmed. 

I think nobody but the admin can answer your question => summoning @kazk (I'm not entirely sure to what repo this should be transfered. I'll let you decide).

Cheers

**Comment by hobovsky** [comment]
Additional libraries can be requested by using an appropriate [issue template](https://github.com/codewars/runner/issues/new/choose) in [codewars/runner](https://github.com/codewars/runner) repository. Please create a package request there by filling the template with all necessary details.
